File: OP/Tema1/KodPrednaska/BankovniUcet.cs
class BankovniUcet
{
    public int cislo;
    private double zustatek;

    public BankovniUcet(int cislo)
    {
        zustatek = 0.0;
        this.cislo = cislo;
    }

    public double VratZustatek()
    {
        return zustatek;
    }
    public void Vloz(double castka)
    {
        // zapis do databaze
        zustatek += castka;
    }

    public void Vyber(double castka)
    { 
        if (castka <= zustatek)
        {
            // zapis do db
            zustatek -= castka;
        }
    }
}

class Prednaska1
{
    static void Main(string[] args)
    {
        BankovniUcet ucet = new BankovniUcet(1);
        ucet.Vloz(2000);
        ucet.Vyber(3000);
        //ucet.zustatek -= 200; // nejde prelozit

        Console.WriteLine($"zustatek: {ucet.VratZustatek()}");
    }
}

File: OP/Tema1/KodPrednaska/bankovni_ucet.c
#include <stdio.h>

struct BankovniUcet
{
    int cislo;
    double zustatek; // nemenit primo, pouzivat metody vloz a vyber
};

void init(struct BankovniUcet* this, int cislo)
{
    this->cislo = cislo;
    this->zustatek = 0.0;
}

void vloz(struct BankovniUcet* this, double castka)
{
    this->zustatek += castka;
}

void vyber(struct BankovniUcet* this, double castka)
{
    if (castka <= this->zustatek)
    {
        this->zustatek -= castka;
    }
}

int main()
{
    // instance na zasobniku
    struct BankovniUcet ucet;
    init(&ucet, 1);
    vloz(&ucet, 2000);
    vyber(&ucet, 200);
    
    // ucet.zustatek -= 200; // jde prelozit

    printf("zustatek: %lf\n", ucet.zustatek);
}
